handlers/register.go

```diff
@@ -54,7 +59,10 @@ func Register(mainRouter *mux.Router, cfg config.Config) {
 	// Set middleware for subrouters
 	rootPaymentRouter.Use(log.Handler, authentication.UserAuthenticationInterceptor)
 	getPaymentRouter.Use(pa.PaymentAuthenticationIntercept)
-	privateRouter.Use(log.Handler, authentication.UserAuthenticationInterceptor, pa.PaymentAuthenticationIntercept)
+	paymentDetailsRouter.Use(interceptors.ElevatedPrivilegesInterceptor)
```

Comment:
if we chain the `pa.PaymentAuthenticationIntercept` interceptor after this one we'll get the DB read for free instead of implementing again in the handler/service in this PR

should also add `log.Handler` as the first interceptor leaving:

`paymentDetailsRouter.Use(log.Handler, interceptors.ElevatedPrivilegesInterceptor, pa.PaymentAuthenticationIntercept)`